curp/src/sync_manager.rs

```diff
@@ -104,97 +108,173 @@ impl<C: Command + 'static> SyncManager<C> {
-                    })
-                    .map(|r| {
-                        match r.into_inner().sync_response {
-                            Some(SyncResponse::Synced(_)) => {
-                                synced_cnt = synced_cnt.overflow_add(1);
-                            }
-                            Some(
-                                SyncResponse::WrongTerm(_)
-                                | SyncResponse::EntryNotEmpty(_)
-                                | SyncResponse::PrevNotReady(_),
-                            ) => {
-                                // todo
-                            }
-                            None => unreachable!("Should contain sync response"),
-                        }
-                    });
-
-                if synced_cnt > max_fail {
-                    for c in cmds_arc.iter() {
-                        if self
-                            .comp_chan
-                            .send(
-                                c.keys(),
-                                SyncCompleteMessage::new(index.numeric_cast(), Arc::clone(c)),
-                            )
-                            .is_err()
-                        {
-                            error!("The comp_chan is closed on the remote side");
-                            return;
-                        }
-                    }
-                    break;
-                }
+                },
+                |r| match r.into_inner().sync_response {
+                    Some(SyncResponse::Synced(_)) => 1,
+                    Some(
+                        SyncResponse::WrongTerm(_)
+                        | SyncResponse::EntryNotEmpty(_)
+                        | SyncResponse::PrevNotReady(_),
+                    ) => 0,
+                    None => unreachable!("Should contain sync response"),
+                },
+                comp_fn,
+                max_fail,
+            )
+            .await
+            {
+                return;
             }
+
+            let connects = self.connects();
+            let _ignore_handle = tokio::spawn(async move {
+                Self::broadcast_and_wait(
+                    connects,
+                    iter::repeat_with(|| Arc::clone(&cmds_arc)),
+                    |(connect, cmds_cloned)| async move {
+                        connect
+                            .commit(CommitRequest::new(
+                                term,
+                                index.numeric_cast(),
+                                cmds_cloned.as_ref(),
+                            )?)
+                            .await
+                    },
+                    |r| match r.into_inner().commit_response {
+                        Some(CommitResponse::Committed(_)) => 1,
+                        Some(CommitResponse::WrongTerm(_) | CommitResponse::PrevNotReady(_)) => 0,
+                        None => unreachable!("Should contain sync response"),
```

Comment:
Should contain commit response

Reply:
unreachable is a panic. The whole process is terminated.